src/main/java/rob/Rob.java

```diff
@@ -0,0 +1,151 @@
+                int taskNum = findTaskNum(input);
+                String markedStr = ui.mark(tasks, taskNum);
+                tasks.getTask(taskNum - 1).markAsDone();
+                storage.saveTasks(tasks);
+                return markedStr;
+            } catch (DukeException e) {
+                System.out.println(e.getMessage());
+                return e.getMessage();
+            }
+        } else if (Objects.equals(command, "unmark")) {
+            try {
+                int taskNum = findTaskNum(input);
+                String unmarkedStr = ui.unmark(tasks, taskNum);
+                tasks.getTask(taskNum - 1).unmark();
+                storage.saveTasks(tasks);
+                return unmarkedStr;
+            } catch (DukeException e) {
+                System.out.println(e.getMessage());
+                return e.getMessage();
+            }
+        } else if (Objects.equals(command, "delete")) {
+            try {
+                int taskNum = findTaskNum(input);
+                String delStr = ui.delete(tasks, taskNum);
+                tasks.removeTask(taskNum - 1);
+                storage.saveTasks(tasks);
+                return delStr;
+            } catch (DukeException e) {
+                System.out.println(e.getMessage());
+                return e.getMessage();
+            }
+        } else if (Objects.equals(command, "find")) {
+            try {
+                String keyword = parser.getFind();
+                List<Task> filteredList = tasks.searchTasks(keyword);
+                TaskList filteredTasks = new TaskList(filteredList);
+                return ui.showList(filteredTasks);
+            } catch (DukeException e) {
+                System.out.println(e.getMessage());
+                return e.getMessage();
+            }
+        } else {
+            try {
+                if (Objects.equals(command, "deadline")) {
+                    String desc = parser.getDesc();
+                    String day = parser.getDay();
+                    tasks.getTasks().add(new Deadline(desc, day));
+                    storage.saveTasks(tasks);
+                } else if (Objects.equals(command, "event")) {
+                    String desc = parser.getDesc();
+                    String from = parser.getFrom();
+                    String to = parser.getTo();
+                    tasks.getTasks().add(new Event(desc, from, to));
+                    storage.saveTasks(tasks);
+                } else if (Objects.equals(command, "todo")) {
+                    String desc = parser.getDesc();
+                    tasks.getTasks().add(new Todo(desc));
+                    storage.saveTasks(tasks);
+                } else {
+                    throw new DukeException("I'm sorry... I don't seem to understand.");
+                }
+                return ui.printText(tasks);
+            } catch (DukeException e) {
+                System.out.println(e.getMessage());
+                return e.getMessage();
```

Comment:
Hi Linette! You could consider using a switch statement to reduce several lines of code here!